code/game/atom/atom_appearance.dm

```diff
@@ -56,6 +56,8 @@
 				continue
 			if(istext(maybe_not_an_atom) || isicon(maybe_not_an_atom))
 				continue
+			if (cached_color_filter && !(appearance_flags & KEEP_TOGETHER))
+				maybe_not_an_atom.add_filter(FILTER_ATOM_PRIORITY_COLOR, 1, cached_color_filter)
```

Comment:
I wonder if we should just outlaw text/icons at this point, not having filter support for them blows.
also should this be made generic or have we not decided

Reply:
we've decided on this one, agree on the former though.